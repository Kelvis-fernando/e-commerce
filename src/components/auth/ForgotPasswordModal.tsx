import { useState } from "react";
import useAuth from "../../hooks/auth/useAuth";
import useToast from "../../hooks/useToast";
import { AuthModalContainer } from "../../styles/auth/loginModal";
import { ForgotPasswordProps } from "../../types/auth";
import Toast from "../Toast";
import TitleAndSubtitle from "../layout/TitleAndSubtitle";
import useLoginModal from "../../hooks/auth/useLoginModal";

const ForgotPasswordModal = ({ onBack }: ForgotPasswordProps) => {
  const { showToast, handleButtonClick, handleCloseToast } = useToast();
  const [errorToast, setErrorToast] = useState(false);
  const { handleGetAuthFormData } = useAuth(handleButtonClick, setErrorToast);
  const { setResetPasswordModalOPen } = useLoginModal();

  return (
    <>
      <AuthModalContainer>
        <TitleAndSubtitle
          title="Esqueceu a sua senha?"
          subtitle="Adicione o seu email para lhe ajudarmos"
        />
        <form onSubmit={(event) => handleGetAuthFormData(event, "forgot")}>
          <label htmlFor="email">E-mail</label>
          <input type="email" name="email" id="email" />
          <button onClick={() => setResetPasswordModalOPen(true)} type="submit">
            Enviar
          </button>
        </form>
        <h2
          onClick={() => {
            if (onBack) return onBack(false);
          }}
        >
          Voltar
        </h2>
      </AuthModalContainer>
      {showToast && (
        <Toast
          message={"Token enviado com sucesso"}
          onClose={handleCloseToast}
        />
      )}
      {errorToast && (
        <Toast
          message={"Erro ao enviar o Token"}
          typeOfToast="error"
          onClose={() => setErrorToast(false)}
        />
      )}
    </>
  );
};

export default ForgotPasswordModal;
