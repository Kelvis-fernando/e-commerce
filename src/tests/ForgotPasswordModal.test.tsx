import React from "react";
import { render } from "@testing-library/react";
import ForgotPasswordModal from "../components/auth/ForgotPasswordModal";

describe("ForgotPasswordModal", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<ForgotPasswordModal />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
