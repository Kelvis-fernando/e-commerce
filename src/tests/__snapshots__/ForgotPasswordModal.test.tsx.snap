// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordModal should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hlQNyq"
  >
    <div
      class="sc-eDvSVe giVUvE"
    >
      <h1>
        Esqueceu a sua senha?
      </h1>
      <h3>
        Adicione o seu email para lhe ajudarmos
      </h3>
    </div>
    <form>
      <label
        for="email"
      >
        E-mail
      </label>
      <input
        id="email"
        name="email"
        type="email"
      />
      <button
        type="submit"
      >
        Enviar
      </button>
    </form>
    <h2>
      Voltar
    </h2>
  </div>
</div>
`;
